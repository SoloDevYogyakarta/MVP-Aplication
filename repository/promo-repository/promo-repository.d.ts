import { PromoInstance } from '../../database/entities/services/promo-entity/promo-entity';
export declare class PromoRepository {
    private readonly logger;
    findAll(query: object, type: string): Promise<PromoInstance[]>;
    findOne(id: number): Promise<PromoInstance>;
}
