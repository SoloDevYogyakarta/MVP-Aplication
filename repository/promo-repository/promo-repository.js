"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var PromoRepository_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromoRepository = void 0;
const common_1 = require("@nestjs/common");
const filter_1 = require("../../utils/filter/filter");
const promo_entity_1 = require("../../database/entities/services/promo-entity/promo-entity");
let PromoRepository = PromoRepository_1 = class PromoRepository {
    constructor() {
        this.logger = new common_1.Logger(PromoRepository_1.name);
    }
    async findAll(query, type) {
        let where = {};
        this.logger.log(PromoRepository_1.name);
        where = (0, filter_1.filter)(query, type);
        return await promo_entity_1.promoEntity.findAll({
            where: where,
            order: [['id', 'ASC']],
        });
    }
    async findOne(id) {
        this.logger.log(PromoRepository_1.name);
        return await promo_entity_1.promoEntity.findOne({ where: { id } });
    }
};
exports.PromoRepository = PromoRepository;
exports.PromoRepository = PromoRepository = PromoRepository_1 = __decorate([
    (0, common_1.Injectable)()
], PromoRepository);
//# sourceMappingURL=promo-repository.js.map