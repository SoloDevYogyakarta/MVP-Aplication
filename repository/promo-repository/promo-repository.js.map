{"version":3,"file":"promo-repository.js","sourceRoot":"","sources":["../../../src/repository/promo-repository/promo-repository.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAAoD;AACpD,sDAAmD;AACnD,6FAGoE;AAG7D,IAAM,eAAe,uBAArB,MAAM,eAAe;IAArB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,iBAAe,CAAC,IAAI,CAAC,CAAC;IAgB7D,CAAC;IAdC,KAAK,CAAC,OAAO,CAAC,KAAa,EAAE,IAAY;QACvC,IAAI,KAAK,GAAG,EAAE,CAAC;QACf,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,iBAAe,CAAC,IAAI,CAAC,CAAC;QACtC,KAAK,GAAG,IAAA,eAAM,EAAC,KAAK,EAAE,IAAI,CAAC,CAAC;QAC5B,OAAO,MAAM,0BAAW,CAAC,OAAO,CAAC;YAC/B,KAAK,EAAE,KAAK;YACZ,KAAK,EAAE,CAAC,CAAC,IAAI,EAAE,KAAK,CAAC,CAAC;SACvB,CAAC,CAAC;IACL,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,iBAAe,CAAC,IAAI,CAAC,CAAC;QACtC,OAAO,MAAM,0BAAW,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;IACtD,CAAC;CACF,CAAA;AAjBY,0CAAe;0BAAf,eAAe;IAD3B,IAAA,mBAAU,GAAE;GACA,eAAe,CAiB3B"}