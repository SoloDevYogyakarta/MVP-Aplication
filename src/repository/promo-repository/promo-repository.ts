import { Injectable, Logger } from '@nestjs/common';
import {
  promoEntity,
  PromoInstance,
} from '../../database/entities/services/promo-entity/promo-entity';

@Injectable()
export class PromoRepository {
  private readonly logger = new Logger(PromoRepository.name);

  async findAll() {
    this.logger.log(PromoRepository.name);
    return await promoEntity.findAll();
  }

  async findOne(id: number): Promise<PromoInstance> {
    this.logger.log(PromoRepository.name);
    return await promoEntity.findOne({ where: { id } });
  }
}
