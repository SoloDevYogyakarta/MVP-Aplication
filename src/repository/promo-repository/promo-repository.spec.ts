import { PromoRepository } from './promo-repository';

describe('PromoRepository', () => {
  let repository: PromoRepository;

  beforeEach(() => {
    repository = new PromoRepository();
  });

  it('should to be defined', () => expect(repository).toBeDefined());

  it('render correctly', () => expect(repository).toMatchSnapshot());
});
