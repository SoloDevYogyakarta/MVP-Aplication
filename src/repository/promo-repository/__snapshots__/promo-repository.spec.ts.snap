// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoRepository render correctly 1`] = `
PromoRepository {
  "logger": Logger {
    "context": "PromoRepository",
    "options": {},
  },
}
`;
